frontend/admin/src/js/components/user/UserTableFilterDialog.tsx

```diff
@@ -250,6 +263,42 @@ const UserTableFilterButton = ({
   const handleSubmit: SubmitHandler<FormValues> = (data) => {
     onSubmit(data);
     setActiveFilters(data);
+    // this state lives in the UserTable component, this step also acts like a formValuesToSubmitData function
+    onFilterChange({
```

Comment:
Do you think this component needs 2 submission handlers? As in could this be moved into the onSubmit function that was already present? Maybe it was just poorly named, but `onSubmit` was the place where I assumed this logic would be injected/added

maybe `onSubmitFilter`? which identifies itself as "for a form" but also "for the filter" like you capture?

Reply:
EDIT: Wait. Nevermind, might have mixed myself up here. gimme a sec :)

Reply:
Bah, I'm fumbling to think how to merge these nicely, but do you see any way that a single function is possible instead of two? I think ideally the processing of formvalues into input type would happen in a single place, but I'm not seeing a clean way to do that. I recall onSubmit is a handler set up one level that doesn't do much, so feels odd to have two

Reply:
if it is too much of a mess to attempt it, then it might not be worth the hassle frankly

Reply:
i'm onboard with that. extracting the processing out of the deep component is the hard part...

but what about just stripping out current `onSubmit` and renaming `onFilterChange` to `onSubmitFilter` (to hint at form submission and show up when grepping for "onSubmit")

Reply:
Oh! Wait... you could also use it to pass your function, and keep the processing out of the component:

```tsx
  const handleFilterSubmit: SubmitHandler<FormValues> = (data) => {
    onFilterChange( //...
    // ...
  };
```